apps/arena/lib/arena/game/player.ex

```diff
@@ -396,6 +391,25 @@ defmodule Arena.Game.Player do
     end
   end
 
+  ## Yes, we are pattern matching on exactly one mechanic. As of time writing we only have one mechanic per skill
+  ## so to simplify my life an executive decision was made to take thas as a fact
+  ## When the time comes to have more than one mechanic per skill this function will need to be refactored, good thing
+  ## is that it will crash here so not something we can ignore
+  defp calculate_duration(%{mechanics: [{:leap, leap}]}, position, direction) do
+    ## TODO: Cap target_position to leap.range
```

Comment:
For what i know the distance is already caped at the leap range right?

Reply:
Nope, there is no backend validation on the actual position you are sending. Meaning that a hacked client could in theory send anything